### PR TITLE
[GHF] Rebase and retry push if fails

### PR DESCRIPTION
**Pull request by malfet**
This fixes a pretty frequently occurring race condition when two
workflows are running concurrently

Resolve this by fetching latest branch and rebating proposed changes on
top of it

Test plan: Simulate the behavior in https://github.com/malfet/deleteme

Fixes #ISSUE_NUMBER

### CODE REVIEW

**Review by seemethere — APPROVED** [review]

File: .github/scripts/gitutils.py [hunk 1]
```diff
@@ -212,11 +217,21 @@ def cherry_pick_commits(self, from_branch: str, to_branch: str) -> None:
...
+        for cnt in range(retry):
+            try:
+                if dry_run:
+                    self._run_git("push", "--dry-run", self.remote, branch)
+                else:
+                    self._run_git("push", self.remote, branch)
+            except RuntimeError as e:
+                # Check if push were rejected because branch is stale
+                if len(e.args) != 1:
+                    raise
```
Comment (bigfootjon) [inline .github/scripts/gitutils.py:230]: This check is kinda sketchy. Is there a better way to do this?

File: .github/scripts/gitutils.py [hunk 2]
```diff
@@ -212,11 +217,21 @@ def cherry_pick_commits(self, from_branch: str, to_branch: str) -> None:
...
+                if re.search("\[rejected\].+\(fetch first\)\n", e.args[0]) is None:
```
Comment (bigfootjon) [inline .github/scripts/gitutils.py:231]: I think you want an `r` string here to fix lint

**Review by bigfootjon — APPROVED** [review]
This plausibly LGTM (with suggested fixes for lint and one nit question)

However, is there a better way to do this? Can we use the concurrency feature of GitHub Actions? https://github.blog/changelog/2021-04-19-github-actions-limit-workflow-run-or-job-concurrency/

File: .github/scripts/gitutils.py (hunk 1 above)
Comment (malfet) [reply .github/scripts/gitutils.py:230]: This limits the scope on what kinds of Exceptions  we are attempting to retry on (i.e. if RuntimeError is raised somewhere else in the stack, but I can probably fold those two checks into one

**Comment by malfet** [comment]
> However, is there a better way to do this? Can we use the concurrency feature of GitHub Actions? https://github.blog/changelog/2021-04-19-github-actions-limit-workflow-run-or-job-concurrency/

This is a neat idea, but I'm not sure if concurrency checks can be enforced across worfkflows (as right now there are three: trymere, revert and syncbranches)

**Pull request closed without merging**